### PR TITLE
Fix/k32w0 shell app

### PR DESCRIPTION
**Pull request by doru91**
### Problem
* shell app was not working on K32W0

### Change overview
* fix linkage error by disabling the ping functionality
* fix deadlock in Thread implementation
* fix printing by adding CR
* This commit relates to #11346

### Testing
* testing using chip-tool

### CODE REVIEW

**Comment by jmartinez-silabs** [comment]
I am confuse. This PR is to apply to the Test event 7 branch ? If you need it in the branch officially should we not just cherry-pick the shell fixes already in master ? 

If you also need other changes for NXP, I think those should first be merged in master and then cherry-picked if need be to the te7 branch

File: examples/shell/nxp/k32w/k32w0/main/main.cpp [hunk 1]
```diff
@@ -134,7 +134,7 @@ extern "C" void main_task(void const * argument)
     }
 
     // cmd_otcli_init();
-    cmd_ping_init();
+    // cmd_ping_init();
```
Comment (andy31415) [inline examples/shell/nxp/k32w/k32w0/main/main.cpp:137]: Should we delete commented out code instead of keeping it in history?

File: src/lib/shell/commands/Dns.cpp [hunk 2]
```diff
@@ -42,23 +42,23 @@ class DnsShellResolverDelegate : public Dnssd::ResolverDelegate
 public:
     void OnNodeIdResolved(const Dnssd::ResolvedNodeData & nodeData) override
     {
-        streamer_printf(streamer_get(), "DNS resolve for " ChipLogFormatX64 "-" ChipLogFormatX64 " succeeded:\n",
+        streamer_printf(streamer_get(), "DNS resolve for " ChipLogFormatX64 "-" ChipLogFormatX64 " succeeded:\r\n",
```
Comment (andy31415) [inline src/lib/shell/commands/Dns.cpp:45]: why did you add \r to shared code? Is it required by the interface? Could it be  resolved at a lower level only for platforms that need it? I am used to printfs only having the '\n' and  woner why  the \r is needed here

**Review by andy31415 — APPROVED** [review]

**Comment by andy31415** [comment]
Adding checkmark because merge  from master. The "\r" seems odd to me since that may not be a correct choice for all platforms (I would argue that probably it was added for windows monitoring rather than linux) however that is in master and probably does  not cause that much harm.

File: src/lib/shell/commands/Dns.cpp (hunk 2 above)
Comment (doru91) [reply src/lib/shell/commands/Dns.cpp:45]: "\r" was added because some terminals are not adding it by default. It shouldn't negatively impact other platforms.

Comment (andy31415) [reply src/lib/shell/commands/Dns.cpp:45]: offline discussion: uart in macos apparently also wants \r.

**Comment by doru91** [comment]
> I am confuse. This PR is to apply to the Test event 7 branch ? If you need it in the branch officially should we not just cherry-pick the shell fixes already in master ?
> 
> If you also need other changes for NXP, I think those should first be merged in master and then cherry-picked if need be to the te7 branch

I made the changes during TE7 and that's why I targeted TE7 branch initially. Later on I also needed it on master branch and opened a PR there. But I agree, In the future I will submit the fix to the master branch and then we can cherry-pick from there.

File: examples/shell/nxp/k32w/k32w0/main/main.cpp (hunk 1 above)
Comment (doru91) [reply examples/shell/nxp/k32w/k32w0/main/main.cpp:137]: Doesn't make any difference to me.
